internal/driver/webui.go

```
@@ -0,0 +1,234 @@
+		reportHandlerError(w, "internal template error", err)
+		return
+	}
+	w.Header().Set("Content-Type", "text/html")
+	w.Write(html.Bytes())
+}
+
+func dotToSvg(dot []byte) ([]byte, error) {
+	cmd := exec.Command("dot", "-Tsvg")
+	out := &bytes.Buffer{}
+	cmd.Stdin, cmd.Stdout, cmd.Stderr = bytes.NewBuffer(dot), out, os.Stderr
+	if err := cmd.Run(); err != nil {
+		return nil, err
+	}
+
+	// Fix dot bug related to unquoted amperands.
+	svg := bytes.Replace(out.Bytes(), []byte("&;"), []byte("&amp;;"), -1)
+
+	// Cleanup for embedding by dropping stuff before the <svg> start.
+	if pos := bytes.Index(svg, []byte("<svg")); pos >= 0 {
+		svg = svg[pos:]
+	}
+	return svg, nil
+}
+
+// disasm generates a web page containing disassembly.
+func (ui *webUI) disasm(w http.ResponseWriter, req *http.Request) {
+	ui.output(w, req, "disasm", "text/plain")
+}
+
+// weblist generates a web page containing disassembly.
+func (ui *webUI) weblist(w http.ResponseWriter, req *http.Request) {
+	ui.output(w, req, "weblist", "text/html")
+}
+
+// output generates a webpage that contains the output of the specified pprof cmd.
+func (ui *webUI) output(w http.ResponseWriter, req *http.Request, cmd, ctype string) {
+	focus := req.URL.Query().Get("f")
+	if focus == "" {
+		fmt.Fprintln(w, "no argument supplied for "+cmd)
+		return
+	}
+	args := []string{cmd, focus}
+	vars := pprofVariables.makeCopy()
+	_, rpt, err := generateRawReport(ui.prof, args, vars, ui.options)
+	if err != nil {
+		reportHandlerError(w, "error generating report", err)
+		return
+	}
+
+	out := &bytes.Buffer{}
+	if err := report.Generate(out, rpt, ui.options.Obj); err != nil {
+		reportHandlerError(w, "error generating report", err)
+		return
+	}
+	w.Header().Set("Content-Type", ctype)
+	io.Copy(w, out)
+}
+
+// getFromLegend returns the suffix of an entry in legend that starts
+// with param.  It returns def if no such entry is found.
+func getFromLegend(legend []string, param, def string) string {
```

Extracting the data from the legend seems a bit fragile. Maybe make pprof fail early in tests if the parameter to extract is not found? Or something else that the tests would detect.